### ISSUE TITLE
Tensorflow 1.4.0 Mac GPU with XLA, CUDA 9, cuDNN 7 and Python 3.6

### ISSUE THREAD

**Issue by nathanielatom** [yaroslavvb/tensorflow-community-wheels#44]
https://github.com/nathanielatom/tensorflow/releases/tag/v1.4.0-mac

**Comment by yaroslavvb** [comment]
I'm curious if this actually lets you use float16 computation, or if Amazon/Nvidia-custom version of TF must be used

The way to test is to do this
```
wget https://raw.githubusercontent.com/yaroslavvb/stuff/master/matmul_benchmark_seq.py
export TF_CPP_MIN_LOG_LEVEL=1
export CUDA_VISIBLE_DEVICES=0
python matmul_benchmark_seq.py --dtype=float16
```

Ideally it would eventually print something like
`8192,87.2323633474`
which means that it got 87 T ops/second for 8192x8192 matrix multiply in fp16


https://medium.com/@yaroslavvb/peak-performance-of-amazon-p3-instances-f2bc48f9ef71
https://devtalk.nvidia.com/default/topic/1025971/container-tensorflow/unexpected-low-fp16-performance-on-p3/

**Comment by nathanielatom** [comment]
No, there's no Float16 computation benefit, just the memory benefit over Float32.

On my 1080 Ti eGPU the max matrix multiply performance is:

dtype | N | TFLOPS
-|---|--------
Float16 | 8192 | 12.69
Float32 | 4096 | 12.79

And the max size of `N` before running out of memory is `9741` for Float32 and `13777` for Float16 as expected. This matches the behaviour of https://github.com/tensorflow/models/blob/master/tutorials/image/mnist/convolutional.py.

I also plotted vs `N` and saw performance drops for non-powers-of-two, but interestingly only for Float32.

![bokeh_plot 56](https://user-images.githubusercontent.com/3137319/32421267-9d197802-c264-11e7-8bd4-4daca599447b.png)

**Comment by rapatel0** [comment]
Try setting these environmental variables. They enable fp16 compute for tf1.4 . It's only worth it however, if you have a volta or tesla p100. Float16 performance is 1/32x on any other cards. 

``` 
ENV TF_FP16_CONV_USE_FP32_COMPUTE=0                                                                                                                                                                                                                                           
ENV TF_FP16_MATMUL_USE_FP32_COMPUTE=0         
```
